src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -489,3 +490,106 @@ def _sql_alter_from(self, table: Table, target_table_key: str, ws_id: int):
             f"('upgraded_from' = '{source}'"
             f" , '{table.UPGRADED_FROM_WS_PARAM}' = '{ws_id}');"
         )
+
+
+class ACLMigrator:
+    def __init__(
+        self,
+        table_crawler: TablesCrawler,
+        grant_crawler: GrantsCrawler,
+        ws: WorkspaceClient,
+        workspace_name: str,
+        backend: SqlBackend,
+        group_manager: GroupManager,
+        migration_status_refresher: MigrationStatusRefresher,
+        principal_grants: PrincipalACL,
+    ):
+        self._tc = table_crawler
+        self._gc = grant_crawler
+        self._backend = backend
+        self._workspace_name = workspace_name
+        self._ws = ws
+        self._group = group_manager
+        self._migration_status_refresher = migration_status_refresher
+        self._seen_tables: dict[str, str] = {}
+        self._principal_grants = principal_grants
+
+    def migrate_acls(self, acl_strategy: list[AclMigrationWhat] | None = None):
```

Comment:
Can you add the return type?

```suggestion
    def migrate_acls(self, acl_strategy: list[AclMigrationWhat] | None = None) -> [?]:
```

Reply:
Make iterables plural

```suggestion
    def migrate_acls(self, acl_strategies: list[AclMigrationWhat] | None = None):
```

Reply:
```suggestion
    def migrate_acls(self, legacy_table_acl: bool = True, principal: bool = True) -> None:
```
let's use kwargs